libraries/Microsoft.Bot.Builder.LanguageGeneration/Templates.cs

```diff
@@ -395,5 +420,66 @@ private void CheckErrors()
                 }
             }
         }
+
+        private string ExtractNameSpace(IList<string> optionStrList)
+        {
+            string result = null;
+            foreach (var optionStr in optionStrList)
+            {
+                if (!string.IsNullOrWhiteSpace(optionStr) && optionStr.Contains("="))
+                {
+                    var index = optionStr.IndexOf('=');
+                    var key = optionStr.Substring(0, index).Trim();
+                    var value = optionStr.Substring(index + 1).Trim();
+                    if (key == "@registerNamespace")
+                    {
+                        result = value;
+                    }
+                }
+            }
+
+            if (result == null)
+            {
+                if (Path.IsPathRooted(this.Id))
+                {
+                    result = Path.GetFileName(this.Id).Split('.')[0];
+                }
+                else
+                {
+                    throw new Exception("namespace is required!");
+                }
+            }
+
+            return result;
+        }
+
+        private IList<string> GetGlobalFunctionTable(IList<string> optionStrList)
+        {
+            var result = new List<string>();
+            foreach (var optionStr in optionStrList)
```

Comment:
We abadoned hungarian notation about 20 years ago, no Str postfix

Reply:
lesson learned, changed to option